modules/sc-mesh-secure-deployment/src/nats/tests/test_mdm_agent.py

```diff
@@ -96,8 +370,518 @@ def test_validate_response_returns_ok_for_200_status_and_empty_payload(self):
         """
         mock_response = MagicMock()
         mock_response.status_code = 200
-        mock_response.text = '{"payload": {"radios": []}}'
+        mock_response.text = '{"payload": {"radios": [{}]}}'
 
-        result = self.mdm_agent._MdmAgent__validate_response(mock_response,
-                                                             mdm_agent.ConfigType.MESH_CONFIG)
-        self.assertEqual(result, 'FAIL')
+        result = self.agent._MdmAgent__validate_response(mock_response,
+                                                         mdm_agent.ConfigType.MESH_CONFIG)
+        self.assertEqual(result, 'OK')
+
+        mock_response.text = '{"payload": {"features": [{}]}}'
+        result = self.agent._MdmAgent__validate_response(mock_response,
+                                                         mdm_agent.ConfigType.FEATURES)
+        self.assertEqual(result, 'OK')
+
+        mock_response.text = '{"payload": {"debug_config": [{}]}}'
+        result = self.agent._MdmAgent__validate_response(mock_response,
+                                                         mdm_agent.ConfigType.DEBUG_CONFIG)
+        self.assertEqual(result, 'OK')
+
+    @patch('mdm_agent.socket.create_connection')
+    @patch('mdm_agent.ssl.create_default_context')
+    @patch('mdm_agent.glob.glob')
+    @patch('mdm_agent.logging.Logger')
+    def test_get_server_cert_type_rsa(self, mock_logger, mock_glob, mock_ssl_context, mock_socket):
```

Comment:
There's also the need to test the reception of a bogus/corrupted certificate :skull:

Reply:
This is testing functionality which tries to detect which certificate needs to be used for MDM server connection.  Actual received certificate validation is missing ( will be added to backlog, needs to be planned what kind of actions needs to be done when corrupted certificate is found ).